docs/registry/upload/_index.md

```diff
@@ -317,18 +334,105 @@ jobs:
 The `build-action` GitHub action relies on a build command that you need to specify in the <file>meta.json</file> file that you created for your module when you first [uploaded it](/registry/upload/#upload-a-custom-module-using-the-cli).
 At the end of your <file>meta.json</file>, add the build configuration:
 
+{{< tabs >}}
+{{% tab name="Single Build File" %}}
+
 ```json {class="line-numbers linkable-line-numbers" data-line="4-7"}
 {
   "module_id": "example-module",
   ...
   "build": {
-    "build": "make module.tar.gz",
-    "arch" : ["linux/amd64", "linux/arm64", "darwin/arm64"]
+    "build": "sh build.sh",          // command that will build your module
+    "path" : "dist/archive.tar.gz",         // optional - path to your built module
+    "arch" : ["linux/amd64", "linux/arm64"] // architecture(s) to build for
   }
 }
 ```
 
-You can test this build configuration by running the following command on your development machine:
+{{%expand "Click to view example build.sh" %}}
+
+```sh { class="command-line"}
+#!/bin/bash
+set -e
+UNAME=$(uname -s)
+
+if [ "$UNAME" = "Linux" ]
+then
+    echo "Installing venv on Linux"
+    sudo apt-get install -y python3-venv
+fi
+if [ "$UNAME" = "Darwin" ]
+then
+    echo "Installing venv on Darwin"
+    brew install python3-venv
+fi
+
+python3 -m venv .venv
+. .venv/bin/activate
+pip3 install -r requirements.txt
+python3 -m PyInstaller --onefile --hidden-import="googleapiclient" src/main.py
+tar -czvf dist/archive.tar.gz dist/main
+```
+
+{{% /expand%}}
+
+{{% /tab %}}
+{{% tab name="Platform Specific" %}}
+
+```json {class="line-numbers linkable-line-numbers" data-line="4-13"}
+{
+  "module_id": "example-module",
+  ...
+  "build": {
+    "arch": {
+          "linux/arm64": {
+            "path" : "dist/archive.tar.gz",               // optional - path to your built module
+            "build": "sh build-linux-arm64.sh" // command that will build your module
+          },
+          "darwin/arm64": {
+            "build": "sh build-darwin-arm64.sh" // command that will build your module
+          }
+        } // architecture(s) to build for
+  }
+}
+```
+
+{{%expand "Click to view example build-linux-arm64.sh" %}}
+
+```sh { class="command-line"}
+#!/bin/bash
+set -e
+
+sudo apt-get install -y python3-venv
```

Comment:
could use bash `if` statement to unify this one and below -- only difference is apt-get vs brew

Reply:
Not addressing as commenting out